File: exp-01/q1-tower-of-hanoi/solution.go
package main

import "fmt"

func towerOfHanoi(n int, from, to, aux byte, count *int) {
	if n == 1 {
		fmt.Printf("Move disk 1 from %c to %c\n", from, to)
		*count++
		return
	}
	towerOfHanoi(n-1, from, aux, to, count)
	fmt.Printf("Move disk %d from %c to %c\n", n, from, to)
	*count++
	towerOfHanoi(n-1, aux, to, from, count)
}

func main() {
	var n int
	fmt.Scan(&n)
	count := 0
	towerOfHanoi(n, 'A', 'C', 'B', &count)
	fmt.Printf("Total moves: %d\n", count)
}

File: exp-01/q1-tower-of-hanoi/Solution.java
import java.util.Scanner;

public class Solution {
    private static int count = 0;

    static void towerOfHanoi(int n, char from, char to, char aux) {
        if (n == 1) {
            System.out.println("Move disk 1 from " + from + " to " + to);
            count++;
            return;
        }
        towerOfHanoi(n - 1, from, aux, to);
        System.out.println("Move disk " + n + " from " + from + " to " + to);
        count++;
        towerOfHanoi(n - 1, aux, to, from);
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        towerOfHanoi(n, 'A', 'C', 'B');
        System.out.println("Total moves: " + count);
        sc.close();
    }
}